core/drivers/crypto/stm32/cipher.c

```diff
@@ -362,6 +362,16 @@ static struct drvcrypt_cipher driver_cipher_saes = {
 
 TEE_Result stm32_register_cipher(enum stm32_cipher_ip_id cipher_ip)
 {
+	void *op = drvcrypt_get_ops(CRYPTO_CIPHER);
+
+	if (op) {
+		EMSG("%s already registered for CRYPTO_CIPHER",
+		     op == &driver_cipher_cryp ? "CRYP peripheral" :
+		     op == &driver_cipher_saes ? "SAES peripheral" :
+		     "Other AES accelerator");
```

Comment:
This is not "AES specific", it's rather "symmetric cipher generic" (CRYP supports AES/DES/3DES).
Prefer also `driver` over `accelerator`, the driver goal may not be to accelerate the processing (e.g. SAES).